### PR TITLE
update NGC docker

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

update docker image for NGC

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

File: dockers/release/Dockerfile [hunk 1]
```diff
@@ -36,7 +36,7 @@ RUN \
         mv pytorch-lightning-*/ pytorch-lightning ; \
         rm *.zip ; \
     fi && \
-    pip install ./pytorch-lightning["extra"] --no-cache-dir && \
+    pip install ./pytorch-lightning["extra","loggers","strategies"] --no-cache-dir --upgrade-strategy only-if-needed && \
```
Comment (carmocca) [inline dockers/release/Dockerfile:39]: Isn't "only-if-needed" the default upgrade strategy anyways?

File: dockers/nvidia/Dockerfile [hunk 2]
```diff
@@ -32,25 +32,24 @@ RUN \
...
         git clone https://github.com/PyTorchLightning/pytorch-lightning.git ; \
         cd pytorch-lightning ; \
-        git checkout ${LIGHTNING_VERSION} ; \
+        git checkout tags/${LIGHTNING_VERSION} ; \
         git submodule update --init --recursive ; \
         cd .. ; \
     fi && \
 # save the examples
-    mv pytorch-lightning/_notebooks notebooks && \
+    mv pytorch-lightning/_notebooks/.notebooks/ notebooks && \
```
Comment (carmocca) [inline dockers/nvidia/Dockerfile:40]: Is this change correct? I don't see the `.notebooks` directory

**Review by carmocca — APPROVED** [review]

File: dockers/nvidia/Dockerfile [hunk 3]
```diff
@@ -32,25 +32,24 @@ RUN \
         rm -rf pytorch-lightning ; \
         git clone https://github.com/PyTorchLightning/pytorch-lightning.git ; \
         cd pytorch-lightning ; \
-        git checkout ${LIGHTNING_VERSION} ; \
+        git checkout tags/${LIGHTNING_VERSION} ; \
```
Comment (akihironitta) [inline dockers/nvidia/Dockerfile:35]: Do we need this prefix `tags/`? It's fine if it works :)
```suggestion
        git checkout ${LIGHTNING_VERSION} ; \
```

File: dockers/nvidia/Dockerfile (hunk 2 above)
Comment (akihironitta) [reply dockers/nvidia/Dockerfile:40]: This directory, isn't it? https://github.com/PyTorchLightning/lightning-tutorials/tree/main/.notebooks

**Review by akihironitta — APPROVED** [review]

File: dockers/release/Dockerfile (hunk 1 above)
Comment (Borda) [reply dockers/release/Dockerfile:39]: good question...

File: dockers/nvidia/Dockerfile (hunk 2 above)
Comment (Borda) [reply dockers/nvidia/Dockerfile:40]: yes, you need to switch to `publication` branch with generated notebooks

File: dockers/nvidia/Dockerfile (hunk 3 above)
Comment (Borda) [reply dockers/nvidia/Dockerfile:35]: I saw git complaining nut I ma fine with the simple way too

File: dockers/release/Dockerfile (hunk 1 above)
Comment (Borda) [reply dockers/release/Dockerfile:39]:
```suggestion
    pip install ./pytorch-lightning["extra","loggers","strategies"] --no-cache-dir && \
```

**Review by kaushikb11 — APPROVED** [review]

**Pull request merged**